src/platform_impl/linux/x11/event_processor.rs

```diff
@@ -1216,3 +1241,19 @@ impl<T: 'static> EventProcessor<T> {
         }
     }
 }
+
+fn is_first_touch(m: &mut HashSet<u64>, id: u64, phase: TouchPhase) -> bool {
+    let is_first = m.is_empty();
```

Comment:
This logic is incorrect except on TouchPhase::Started. You also don't actually need a hashmap here.

Reply:
Fixed again. Hopefully.

Reply:
Also, I changed the code from using a `HashSet` to a simple count of concurrent running touch events. As long as winit doesn't receive a duplicate `Ended` event, it should work as well as the `HashSet`.